accepted/0000-subscription-matching-in-public-clouds.md

```diff
@@ -0,0 +1,194 @@
+
+instances = list_instances(compute, project, zone)
+for i in instances:
+    print(i)
+```
+
+The response contains the "instance name" as well as the "instance id".
+
+### A generic JSON-file module:
+
+Currently, there is already a "File based" plugin for `virtual-host-gatherer` which allows to import instances from a custom provided JSON file. According to the existing documentation, the aim of this plugin is to import VMware instances when there is no access from the SUSE Manager server to the VMware.
+
+We can already use this plugin for importing virtual instances from the Public Cloud using a tailored JSON file like the one from the above example, so it would be worth to generalize the SUSE Manager documentation to expose this plugin as a general virtual instances importer and not as a VMware-specific.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## Gathering the UUID (instance id) from virtual instances
+
+For AWS and GCE, the instance id that is returned from the API is not actually the real "uuid" from the instance. On Azure, the instance id is an UUID but not necessary correspond with the SMBIOS "uuid" value we get from Salt.
+
+An easy approach here would be to use the "Instance ID" (instead of smbios uuid) when registering a system which is a public cloud virtual instance. Salt currently does not provide the instance id as part of the grains but it would be really easy to provide a custom grain at the time of registration that would expose the "instance id" as part of the grains only when the system is an EC2, GCE or Azure instance. [Example here](https://gist.github.com/meaksh/1ed58ece0f26ce27a8445985de9ad6a2)
+
+This way, doing some minor fixes on the Java side ([example here](https://github.com/meaksh/uyuni/commit/03d88550dd87d22f3fabd25cebd7c23432285a3c)), we could easily use the "instance-id" as "UUID" for the registered system and automatically match it with the data provided by the `virtual-host-gatherer` plugin (which does not include "uuid" but instance id).
+
+In case of systems that are already registered in SUSE Manager using a smbios "uuid", if the new "instance_id" grain is there, it should be enough with scheduling a "Hardware Refresh" action to reflect the new "instance_id" grain value as the "uuid" for that system.
+
+# Alternatives
+[alternatives]: #alternatives
+
+### Use special "System Grouping" and do not use `virtual-host-gatherer`:
+This alternative approach will be based on implementing an special "System Groups" where Administrators could define their virtualization enviroments and set the systems that belong to those groups.
+
+Pros:
+- All virtual environments are defining via grouping. No need to actually reach the Public Clouds, no external APIs.
+
+Cons:
+- These virtualization groups need to be defined manually. No way of automation.
+- Refreshing the state of a virtualization group also needs to be done manually.
+- Require new DB tables and UI for managing those groups.
```

Comment:
This is all correct. A note from my side: My feeling is, that no customer would "maintain" these data.